### PR TITLE
Hs 10 delete meeting

### PR DESCRIPTION
**Pull request by reut-soficoop**
to dev

### CODE REVIEW

File: src/helpers/queries.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 export function getMeetings(date) {
   return `query getMeetings{
-  meetings(where: {date_gt: "${date.toISOString()}"} sort: "date"){
+  meetings(where: {date_gt: "${date.toISOString()}", isHidden: "false"} sort: "date"){
```
Comment (avirankatzsofi) [inline src/helpers/queries.js]: shouldn't it be `isHidden: false`? (without " ")

File: src/helpers/queries.js [hunk 2]
```diff
@@ -162,11 +162,14 @@ export function getCommentsByPlan(id) {
 
 export function getCommitteeMeetings(committeeIds) {
   return `query getCommitteeMeetings {
-    meetings(where: { committee_in: ${JSON.stringify(committeeIds)} }) {
+    meetings(where: { committee_in: ${JSON.stringify(
+      committeeIds
+    )}, isHidden: "false"}) {
```
Comment (avirankatzsofi) [inline src/helpers/queries.js]: shouldn't it be `isHidden: false`? (without " ")

File: src/helpers/mutations.js [hunk 3]
```diff
@@ -206,6 +207,23 @@ export const emailMeeting = `mutation emailMeeting($id:ID!) {
   }
 }`;
 
+export const hideMeeting = `mutation hideMeeting(
+  $id: ID!
+) {
+  updateMeeting(
```
Comment (avirankatzsofi) [inline src/helpers/mutations.js]: please use `updateMyMeeting` instead, for security reasons

File: src/views/Meeting.vue [hunk 4]
```diff
@@ -202,5 +224,18 @@ export default class Meeting extends Vue {
...
     await this.fetchManagableMeetings();
   }
+
+  async deleteMeeting() {
+    try {
+      const meeting = { id: this.meeting.id };
+      await makeGqlRequest(hideMeeting, meeting, this.jwt);
+    } catch (e) {
+      console.error(e);
+      this.errorOccurred = true;
```
Comment (avirankatzsofi) [inline src/views/Meeting.vue:238]: does `errorOccurred` has a use?

File: tests/unit/meeting.spec.js [hunk 5]
```diff
@@ -198,9 +198,24 @@ describe("Meeting.vue", () => {
...
+    expect(wrapper.vm.isMeetingEditable).toBeTruthy();
+    expect(wrapper.text()).toContain("מחיקת ישיבה");
+    expect(wrapper.text()).toContain("עריכת ישיבה");
+  });
+  it("should open delete confiramtion popup", () => {
+    const deleteButton = wrapper.find(".delete-button");
+    deleteButton.trigger("click");
+    expect(wrapper.text()).toContain("האם ברצונך למחוק את הישיבה?");
+  });
+  it("meeting is editable and can be edited or removed", () => {
```
Comment (avirankatzsofi) [inline tests/unit/meeting.spec.js]: duplicate test of `it("dispaly edit and remove button for a meeting"`

File: tests/unit/meeting.spec.js [hunk 6]
```diff
@@ -198,9 +198,24 @@ describe("Meeting.vue", () => {
     expect(wrapper.text()).not.toContain("ישיבה מספר");
     expect(wrapper.text()).toContain("כותרת ישיבה");
   });
-  it("meeting is not editable", () => {
-    expect(wrapper.vm.isMeetingEditable).toBeFalsy();
-    expect(wrapper.text()).not.toContain("עריכת ישיבה");
+  // it("meeting is not editable", () => {
```
Comment (avirankatzsofi) [inline tests/unit/meeting.spec.js]: Could you make a "not editable" test?

**Review by avirankatzsofi — CHANGES_REQUESTED** [review]

File: src/helpers/queries.js (hunk 2 above)
Comment (reut-soficoop) [reply src/helpers/queries.js]: It's not working without the quotes

File: src/helpers/mutations.js (hunk 3 above)
Comment (reut-soficoop) [reply src/helpers/mutations.js]: updated inside the branch

File: src/views/Meeting.vue (hunk 4 above)
Comment (reut-soficoop) [reply src/views/Meeting.vue:238]: added a use

File: tests/unit/meeting.spec.js (hunk 6 above)
Comment (reut-soficoop) [reply tests/unit/meeting.spec.js]: Let's talk about it. I didn't succeed to do both tests -  editable and none editable

File: src/helpers/queries.js (hunk 2 above)
Comment (avirankatzsofi) [reply src/helpers/queries.js]: I just remembered - we need to use an operator..
So it should work like this: `isHidden_ne: true` 
(which means "isHidden not equals true")

File: src/helpers/queries.js (hunk 1 above)
Comment (avirankatzsofi) [reply src/helpers/queries.js]: I just remembered - we need to use an operator..
So it should work like this: isHidden_ne: true
(which means "isHidden not equals true")

Comment (reut-soficoop) [reply src/helpers/queries.js]: updated in branch

File: src/helpers/queries.js (hunk 2 above)
Comment (reut-soficoop) [reply src/helpers/queries.js]: updated in branch

**Review by avirankatzsofi — APPROVED** [review]

**Pull request merged**